core/thread/identity.go

```diff
@@ -167,8 +187,30 @@ func NewToken(issuer crypto.PrivKey, key PubKey) (tok Token, err error) {
 	return Token(str), nil
 }
 
-// Validate returns non-nil if token was issued by issuer.
-// If token is present and valid, the embedded public key is returned.
+// PubKey returns the public key encoded in the token.
+// Note: This does NOT verify the token.
+func (t Token) PubKey() (PubKey, error) {
```

Comment:
This is used to pluck the pub key from a `thread.Token` to be used in the collection `WriteValidator` and `ReadFilter` functions.